src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -129,33 +267,437 @@ public ResultSet getSchemas() throws SQLException {
 
     @Override
     public ResultSet getCatalogs() throws SQLException {
-        throw new SQLFeatureNotSupportedException("TODO");
+        MongoJsonSchema resultSchema = MongoJsonSchema.createEmptyObjectSchema();
+        resultSchema.addRequiredScalarKeys(new Pair<>(TABLE_CAT, BSON_STRING_TYPE_NAME));
+
+        // All fields in this result set are nested under the bottom namespace.
+        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
+        botSchema.properties.put(BOT_NAME, resultSchema);
+
+        BsonExplicitCursor c =
+                new BsonExplicitCursor(
+                        this.conn
+                                .mongoClient
+                                .listDatabaseNames()
+                                .into(new ArrayList<>())
+                                .stream()
+                                .sorted()
+                                .map(
+                                        dbName ->
+                                                createBottomBson(
+                                                        new BsonElement(
+                                                                TABLE_CAT, new BsonString(dbName))))
+                                .collect(Collectors.toList()));
+
+        return new MongoSQLResultSet(null, c, botSchema);
+    }
+
+    /**
+     * liftSQLException tries to execute a Supplier that may throw a RuntimeException that wraps a
+     * SQLException as the cause. If such an exception is encountered, the inner SQLException is
+     * thrown. Otherwise, the Supplier executes as expected.
+     *
+     * <p>This method is intended for use with Stream API methods that wrap SQLExceptions in
+     * RuntimeExceptions to appease the compiler. We want to propagate those SQLExceptions all the
+     * way out to the caller so this method is a way of recovering them.
+     *
+     * @param f The Supplier function to execute. This function may throw a RuntimeException that
+     *     wraps a SQLException.
+     * @param <T> The return type of the Supplier.
+     * @return The return value of the Supplier.
+     * @throws SQLException If a RuntimeException that wraps a SQLException is caught.
+     */
+    private static <T> T liftSQLException(Supplier<T> f) throws SQLException {
+        try {
+            return f.get();
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof SQLException) {
+                throw (SQLException) e.getCause();
+            }
+            throw e;
```

Comment:
Will there be a case where the `RuntimeException` has useful information and it wraps a SQLException?  I'm curious what the type of the RuntimeException is that wraps the SQLException.

Reply:
There never will be (unless some code not owned by us catches `SQLException`s and wraps them in `RuntimeExceptions`, which I suspect is not the case).

The type that wraps the `SQLException` _is_ `RuntimeException` (which is a concrete class). In a few places in this PR, we explicitly catch `SQLException`s in streams and immediately wrap them in and throw runtime exceptions. All places that happen are surrounded by this `liftSQLException` method. So the only time we'll hit this code is if we specifically intended to.